### PR TITLE
Added Filter Response Functionality 

### PR DESCRIPTION
**Pull request by allansifuna**
I Added filter response functionality in accordance to https://restcountries.eu/#filter-response.

### CODE REVIEW

File: restcountries/base.py [hunk 1]
```diff
@@ -7,20 +7,35 @@ class RestCountryApi:
     QUERY_SEPARATOR = ','
 
     @classmethod
-    def _get_country_list(cls, resource, term=''):
+    def _get_country_list(cls, resource, term='', filters=None):
         """Takes a resource and a search term and return a list of countries or a country.
 
         :param resource - resource to create the URL
         :param term - search term provided by the user of this package
+        :param filters - alist of fields to filter the output of the request to include only the specified fields.
```
Comment (SteinRobert) [inline restcountries/base.py:15]: `alist` should probably be `a list`. This typo is present in every docstring.

File: restcountries/base.py [hunk 2]
```diff
@@ -7,20 +7,35 @@ class RestCountryApi:
...
+            filters_uri_string = "fields={}".format(filter_string)
+
         # build uri
         if term and not resource.endswith("="):
             # add the forward slash only when there is a term
             # and it is not specifying the value part of a query string
             term = "/{}".format(term)
 
-        uri = "{}{}{}".format(cls.BASE_URI, resource, term)
+        if filters and resource.endswith("="):
```
Comment (SteinRobert) [inline restcountries/base.py:29]: I believe there a better way to write this. Something like the following maybe?
```python
uri = "{}{}{}".format(cls.BASE_URI, resource, term)
prefix = "?"
if "?" in uri:
   prefix = "&"
uri += "{}{}".format(prefix, filters_uri_string)
```

**Comment by SteinRobert** [comment]
@allansifuna thank you very much for your PR. I really appreciate it!

I have added a few remarks to the code. Furthermore it would be very helpful if you added tests for the new functionality as well as an example in the README.md

**Review by SteinRobert — CHANGES_REQUESTED** [review]
Hi @allansifuna! Thank you very much for you PR! I really appreciate it! I added some remarks to the code. Furthermore - could you provide tests and add an example for the new functionality to the README.md? :)

File: restcountries/base.py (hunk 1 above)
Comment (allansifuna) [reply restcountries/base.py:15]: On it @SteinRobert  let me push the changes

**Comment by SteinRobert** [comment]
@allansifuna thank you very much for the PR! :)

**Pull request merged**